packages/react-native/ReactCommon/react/runtime/TimerManager.cpp

```diff
@@ -258,49 +258,36 @@ void TimerManager::attachGlobals(jsi::Runtime& runtime) {
   }
 
   runtime.global().setProperty(
-      runtime,
-      "setTimeout",
-      jsi::Function::createFromHostFunction(
-          runtime,
-          jsi::PropNameID::forAscii(runtime, "setTimeout"),
-          3, // Function, delay, ...args
-          [this](
-              jsi::Runtime& rt,
-              const jsi::Value& thisVal,
-              const jsi::Value* args,
-              size_t count) {
+    runtime,
+    "setTimeout",
+    jsi::Function::createFromHostFunction(
+        runtime,
+        jsi::PropNameID::forAscii(runtime, "setTimeout"),
+        3, // Function, delay, ...args
+        [this](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count) {
             if (count == 0) {
-              throw jsi::JSError(
-                  rt,
-                  "setTimeout must be called with at least one argument (the function to call).");
+                throw jsi::JSError(rt, "setTimeout must be called with at least one argument (the function to call).");
             }
 
-            if (!args[0].isObject() || !args[0].asObject(rt).isFunction(rt)) {
-              throw jsi::JSError(
-                  rt, "The first argument to setTimeout must be a function.");
-            }
-            auto callback = args[0].getObject(rt).getFunction(rt);
+            auto callback = args[0].isObject() && args[0].asObject(rt).isFunction(rt)
+                                ? args[0].getObject(rt).getFunction(rt)
+                                : jsi::Function::createFromHostFunction(
+                                      rt,
+                                      jsi::PropNameID::forAscii(rt, "dummyFunction"),
+                                      0,
+                                      [](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count) {
+                                          return jsi::Value::undefined();
+                                      });
 
-            if (count > 1 && !args[1].isNumber() && !args[1].isUndefined()) {
-              throw jsi::JSError(
-                  rt,
-                  "The second argument to setTimeout must be a number or undefined.");
-            }
-            auto delay =
-                count > 1 && args[1].isNumber() ? args[1].getNumber() : 0;
+            auto delay = (count > 1 && args[1].isNumber()) ? args[1].getNumber() : 0;
```

Comment:
We should also check for `delay < 0` and assign 0 in that case as well. You can use `std::max(args[1].getNumber(), 0)` in this same expression :)